extension/src/Extension.ts

```diff
@@ -312,7 +318,21 @@ export class Extension {
     private async restartServer(): Promise<void> {
         if (this.server.isReady()) {
```

Comment:
Logic here may have some problem.

When the restartServer() is invoked, it may happen that builds are cancelled but the server is not restart.

I think you should let the restartServer() do a pure thing about restart. Which is said, ask if want to reload first, if yes, call the method `restartServer()`. Another benefit of doing this yes that we can send the info when settings are changed, to know how many users will change those settings during developing.

Reply:
Yeah I think I should put `await this.client.cancelBuilds()` inside `if (selection === action)` statement.